packages/web3-utils/src/event_emitter.ts

```diff
@@ -0,0 +1,104 @@
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { EventEmitter as NodeEventEmitter } from 'events';
+
+type Callback = (params: any) => void | Promise<void>;
+
+export class InBrowserEventEmitter extends EventTarget {
```

Comment:
Where is this implementation comming from? Why not just use https://www.npmjs.com/package/events ?

Reply:
Hi @mpetrunic,
This implementation comes from Stackoverflow with the help of ChatGPT :smile: .
Because `events` module is only a `node` module that does not exist for the browsers. There is a need to use something different for the browsers. And this class simply uses the class `EventTarget` available at the browsers and adds more methods to it which we need in our code.
The other alternative to this solution is to use a 3rd party library like: https://www.npmjs.com/package/eventemitter, https://www.npmjs.com/package/events and https://www.npmjs.com/package/eventemitter3. However, I did not want to add an additional dependency.
However, if you have a better approach, kindly advise.
Thanks,

Reply:
Event emmitters are tricky beasts, easy to cause memory leaks or annoying bugs. I would rely on some solid implementation like https://www.npmjs.com/package/eventemitter3

Reply:
This lib is ~70kb unpack size, for now its good to use dependable lib but also create issue as lib optimisation in future we may attempt reducing lib size then.

Reply:
Thanks @mpetrunic and @jdevcs, for your valid concern and good suggestions.
Because these MR changes still rely on node `EventEmitter`, when it is a node environment, there are no changes for node js users. And for the browser, it depends on the available class `EventTarget` by just wrapping it to use it similar to node `EventEmitter`. So, I see this approach as stable. And, actually, our end-to-end tests pass. But I still need to write unit tests.
However, as you suggested, I can use `[eventemitter3](https://www.npmjs.com/package/eventemitter3)` for now and create a task to write our own Event Emitter later. This is in case you still see the proposed solution in the MR is not good enough even after writing some unit tests. What do you think?

[There are failed tests for Firefox. But they are not related to the changes in this MR as they exist in other MRs as well]